File: UI/eventba_admin/lib/models/user_question/user_question.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_question.g.dart';

@JsonSerializable()
class UserQuestion {
  final String id;
  final String userId;
  final String receiverId;
  final String question;
  final String? answer;
  final bool isQuestionForAdmin;
  final bool isAnswered;
  final String askedAt;
  final String? answeredAt;
  final String createdAt;
  final String updatedAt;
  final String? userEmail;
  final String? userFullName;

  UserQuestion({
    required this.id,
    required this.userId,
    required this.receiverId,
    required this.question,
    this.answer,
    required this.isQuestionForAdmin,
    required this.isAnswered,
    required this.askedAt,
    this.answeredAt,
    required this.createdAt,
    required this.updatedAt,
    this.userEmail,
    this.userFullName,
  });

  factory UserQuestion.fromJson(Map<String, dynamic> json) =>
      _$UserQuestionFromJson(json);

  Map<String, dynamic> toJson() => _$UserQuestionToJson(this);
}

File: EventBa.Services/Database/UserQuestion.cs
namespace EventBa.Services.Database;

public partial class UserQuestion
{
    public Guid Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public Guid UserId { get; set; }
    public Guid ReceiverId { get; set; }
    public Guid? EventId { get; set; }
    public string Question { get; set; } = null!;
    public string? Answer { get; set; }
    public bool IsQuestionForAdmin { get; set; }
    public bool IsAnswered { get; set; }
    public DateTime AskedAt { get; set; }
    public DateTime? AnsweredAt { get; set; }
    public virtual User Receiver { get; set; } = null!;
    public virtual User User { get; set; } = null!;
    public virtual Event? Event { get; set; }
}
